src/pynwb/__init__.py

```diff
@@ -245,6 +245,19 @@ def __init__(self, **kwargs):
                 manager = get_manager()
         super(NWBHDF5IO, self).__init__(path, manager=manager, mode=mode, file=file_obj, comm=comm)
 
+    @docval(*get_docval(_HDF5IO.read))
+    def read(self, **kwargs):
+        try:
+            return super(NWBHDF5IO, self).read(**kwargs)
+        except ValueError as e:
+            built = next(iter(self._HDF5IO__built.values()))
```

Comment:
@ajtritt and I suggest checking for `neurodata_type`. If it does not exist, then the file is not 2.0+.

To report the NWB 1.x version, it would be better to open the h5py.File and look for the root-level dataset 'nwb_version' (around 1.0.4) or the root-level dataset 'neurodata_version' (1.0.0).